File: hooks/scripts/_archived_ts/notifications/providers/telegram.ts
// Telegram notification provider — uses Telegram Bot API with Markdown formatting
import * as path from "node:path";
import { send } from "../lib/sender.ts";
import type { SendResult } from "../lib/sender.ts";

export interface HookInput {
  hook_event_name?: string;
  cwd?: string;
  session_id?: string;
  agent_type?: string;
}

function getTimestamp(): string {
  const now = new Date();
  const pad = (n: number) => String(n).padStart(2, "0");
  return `${now.getFullYear()}-${pad(now.getMonth() + 1)}-${pad(now.getDate())} ` +
    `${pad(now.getHours())}:${pad(now.getMinutes())}:${pad(now.getSeconds())}`;
}

function formatMessage(input: HookInput): string {
  const hookType = input.hook_event_name || "unknown";
  const projectDir = input.cwd || "";
  const sessionId = input.session_id || "";
  const projectName = projectDir ? path.basename(projectDir) : "unknown";
  const timestamp = getTimestamp();
  const sessionDisplay = sessionId ? `${sessionId.slice(0, 8)}...` : "N/A";

  switch (hookType) {
    case "Stop":
      return `🚀 *Project Task Completed*\n\n📅 *Time:* ${timestamp}\n📁 *Project:* ${projectName}\n🆔 *Session:* ${sessionDisplay}\n\n📍 *Location:* \`${projectDir}\``;

    case "SubagentStop": {
      const agentType = input.agent_type || "unknown";
      return `🤖 *Project Subagent Completed*\n\n📅 *Time:* ${timestamp}\n📁 *Project:* ${projectName}\n🔧 *Agent Type:* ${agentType}\n🆔 *Session:* ${sessionDisplay}\n\nSpecialized agent completed its task.\n\n📍 *Location:* \`${projectDir}\``;
    }

    case "AskUserPrompt":
      return `💬 *User Input Needed*\n\n📅 *Time:* ${timestamp}\n📁 *Project:* ${projectName}\n🆔 *Session:* ${sessionDisplay}\n\nClaude is waiting for your input.\n\n📍 *Location:* \`${projectDir}\``;

    default:
      return `📝 *Project Code Event*\n\n📅 *Time:* ${timestamp}\n📁 *Project:* ${projectName}\n📋 *Event:* ${hookType}\n🆔 *Session:* ${sessionDisplay}\n\n📍 *Location:* \`${projectDir}\``;
  }
}

export const telegram = {
  name: "telegram",
  isEnabled: (env: Record<string, string>) => !!(env.TELEGRAM_BOT_TOKEN && env.TELEGRAM_CHAT_ID),
  send: async (input: HookInput, env: Record<string, string>): Promise<SendResult> => {
    const message = formatMessage(input);
    const url = `https://api.telegram.org/bot${env.TELEGRAM_BOT_TOKEN}/sendMessage`;
    return send("telegram", url, {
      chat_id: env.TELEGRAM_CHAT_ID,
      text: message,
      parse_mode: "Markdown",
      disable_web_page_preview: true,
    });
  },
};

File: hooks/scripts/internal/notify/telegram.go
// telegram.go: Telegram Bot API notification provider.
package notify

import (
	"fmt"
	"path/filepath"
	"time"
)

// Telegram is the Telegram Bot API provider.
var Telegram = Provider{
	Name: "telegram",
	IsEnabled: func(env map[string]string) bool {
		return env["TELEGRAM_BOT_TOKEN"] != "" && env["TELEGRAM_CHAT_ID"] != ""
	},
	Send: func(input map[string]interface{}, env map[string]string) (*SendResult, error) {
		token := env["TELEGRAM_BOT_TOKEN"]
		chatID := env["TELEGRAM_CHAT_ID"]
		url := fmt.Sprintf("https://api.telegram.org/bot%s/sendMessage", token)
		msg := formatTelegramMessage(input)
		body := map[string]interface{}{
			"chat_id":                  chatID,
			"text":                     msg,
			"parse_mode":               "Markdown",
			"disable_web_page_preview": true,
		}
		return Send("telegram", url, body, nil), nil
	},
}

// getTimestamp returns a formatted timestamp string (YYYY-MM-DD HH:MM:SS).
func getTimestamp() string {
	now := time.Now()
	return fmt.Sprintf("%04d-%02d-%02d %02d:%02d:%02d",
		now.Year(), now.Month(), now.Day(),
		now.Hour(), now.Minute(), now.Second())
}

// formatTelegramMessage builds the Markdown message text for a given hook input.
func formatTelegramMessage(input map[string]interface{}) string {
	hookType := strField(input, "hook_event_name", "unknown")
	cwd := strField(input, "cwd", "")
	sessionID := strField(input, "session_id", "")
	projectName := "unknown"
	if cwd != "" {
		projectName = filepath.Base(cwd)
	}
	ts := getTimestamp()
	sessionDisplay := "N/A"
	if len(sessionID) >= 8 {
		sessionDisplay = sessionID[:8] + "..."
	} else if sessionID != "" {
		sessionDisplay = sessionID
	}

	switch hookType {
	case "Stop":
		return fmt.Sprintf(
			"🚀 *Project Task Completed*\n\n📅 *Time:* %s\n📁 *Project:* %s\n🆔 *Session:* %s\n\n📍 *Location:* `%s`",
			ts, projectName, sessionDisplay, cwd)
	case "SubagentStop":
		agentType := strField(input, "agent_type", "unknown")
		return fmt.Sprintf(
			"🤖 *Project Subagent Completed*\n\n📅 *Time:* %s\n📁 *Project:* %s\n🔧 *Agent Type:* %s\n🆔 *Session:* %s\n\nSpecialized agent completed its task.\n\n📍 *Location:* `%s`",
			ts, projectName, agentType, sessionDisplay, cwd)
	case "AskUserPrompt":
		return fmt.Sprintf(
			"💬 *User Input Needed*\n\n📅 *Time:* %s\n📁 *Project:* %s\n🆔 *Session:* %s\n\nClaude is waiting for your input.\n\n📍 *Location:* `%s`",
			ts, projectName, sessionDisplay, cwd)
	default:
		return fmt.Sprintf(
			"📝 *Project Code Event*\n\n📅 *Time:* %s\n📁 *Project:* %s\n📋 *Event:* %s\n🆔 *Session:* %s\n\n📍 *Location:* `%s`",
			ts, projectName, hookType, sessionDisplay, cwd)
	}
}
